### PR TITLE
 services route

### LINKED ISSUE #7: GET/ services route
#### services  Route
get all services in JSON format from the database

--

#### Method : GET 

---

#### Main Things : 
- [ ] Database
- [ ] Response

---

#### Database :
Use database query to fetch all the  data  related to services 

---
#### Response : 
Response may be success or error :
* Error :
```javascript
    {
        status:400,
        messgae:'Bad reqeust'
    }
```
* Success
```javascript
    {
        status:200,
        data: [{id: 1, name: "hair"}, {id: 2, name: "nail"}, {id: 3, name: "skin care"}, {id: 4, name: "makeup"}] , 
    }
```

### PR DESCRIPTION
**Pull request by imansedky5**
fix #7 

- fixed **dotenv** call in app.js
 - created commonRouter to include unprotected routes
- created **servicesHandler**  and   **getServices** query  to get all services from DB
- transferred **jest** and **super** test into **dependencies** depends on **es-lint** warning
- created tests for **getServices** query and` /api/services `route as well in **getAllServices.test.js**
- removed `--runInBand --detectOpenHandles` flags from **test script** in **package.json**

### CODE REVIEW

**Review by khamis-hesham — DISMISSED** [review]

**Review by yasmeenattallah — DISMISSED** [review]

**Review by AdhamHaisami — DISMISSED** [review]

File: server/controller/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const { servicesHandler } = require('./user/index');
```
Comment (mFlaifel) [inline server/controller/index.js:1]: please move unprotected route to common folder

File: server/routes/index.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 const express = require('express');
 
+const { servicesHandler } = require('../controller');
+
 const router = express.Router();
 
-router.get('/hi', (req, res) => {
-  res.send('hi');
-});
+router.get('/services', servicesHandler);
```
Comment (mFlaifel) [inline server/routes/index.js:7]: please create a common.js file in the routes folder and add your route to it.

common.js
```js
const commonRouter = require('express').Router();

const {
  serviceHandler
} = require('../controllers');

commonRouter.get('/services', serviceHandler);

module.exports = commonRouter;
```
then import it into the index.js
```js
const commonRouter = require('./common');

router.use(commonRouter);
```

**Review by mFlaifel — COMMENTED** [review]
good job, but don't forget to rename your test file to something like `getAllServices.test.js`

**Comment by imansedky5** [comment]
> good job, but don't forget to rename your test file to something like `getAllServices.test.js`

done

**Review by khamis-hesham — APPROVED** [review]

**Pull request merged**